### ISSUE TITLE
Failure read graph exported from R

### ISSUE THREAD

**Issue by cccfran** [igraph/igraph#1150]
I built graph in R with NA edge weights and exported as net and graphml format but failed to import (following the sample codes `graphml.c` and `pajek2.c`) with unreadable error code. I checked the exported .graphml can be read by R and python but not .net (though pajek is able to). I suspect failing to read the exported .net in R is due to the format of the vertices in .net file. Not sure why neither can be read in C. Thanks!

**System setup as follows:**
`Linux node 3.10.0-957.1.3.el7.x86_64 #1 SMP Thu Nov 15 17:36:42 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux`

**.net file as follows:**
```
*Vertices 7
*Arcs
1 6 NaN
2 6 NaN
3 7 100
4 7 191
5 7 9
```

**.graphml file as follows:**
```
<?xml version="1.0" encoding="UTF-8"?>
<graphml xmlns="http://graphml.graphdrawing.org/xmlns"
         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:schemaLocation="http://graphml.graphdrawing.org/xmlns
         http://graphml.graphdrawing.org/xmlns/1.0/graphml.xsd">
<!-- Created by igraph -->
  <key id="v_name" for="node" attr.name="name" attr.type="string"/>
  <key id="e_share" for="edge" attr.name="share" attr.type="double"/>
  <key id="e_weight" for="edge" attr.name="weight" attr.type="double"/>
  <graph id="G" edgedefault="directed">
    <node id="n0">
      <data key="v_name">601027</data>
    </node>
    <node id="n1">
      <data key="v_name">610558</data>
    </node>
    <node id="n2">
      <data key="v_name">4966328</data>
    </node>
    <node id="n3">
      <data key="v_name">549871</data>
    </node>
    <node id="n4">
      <data key="v_name">1039879</data>
    </node>
    <node id="n5">
      <data key="v_name">1002171</data>
    </node>
    <node id="n6">
      <data key="v_name">1002177</data>
    </node>
    <edge source="n0" target="n5">
    </edge>
    <edge source="n1" target="n5">
    </edge>
    <edge source="n2" target="n6">
      <data key="e_share">33.3333</data>
      <data key="e_weight">100</data>
    </edge>
    <edge source="n3" target="n6">
      <data key="e_share">63.6667</data>
      <data key="e_weight">191</data>
    </edge>
    <edge source="n4" target="n6">
      <data key="e_share">3</data>
      <data key="e_weight">9</data>
    </edge>
  </graph>
</graphml>

```

**Error code as follows**
```
 ``|0 ``?TT@T@DDP?td?@?88@8@@@
                     ?
                      @?
                        @44Q?tdR?td``/lib64/ld-linux-x86-64.so.2GNU GNU ?Lj?x???[
                                                                                 ???????a?? @)BE???||?Q~????qX?????x`??C????? ?|`0`+?`?
                            h@?|`%4
                                   @libigraph.so.0__gmon_start___init_finiigraph_cattribute_tableigraph_i_set_attribute_tableigraph_read_graph_pajekigraph_vcountigraph_ecountigraph_is_directedigraph_destroylibc.so.6fopenputsputcharprintffclose__libc_start_main_edata__bss_start_endGLIBC_2.2.5?ui	
?`` `(`0`8`@`H`PX`      ``                                                              ?`
h`
  p`
    H?H? H??t??H???5 ?% @?% h??????% h??????% h??????% h?????% h?????% h?????% h?????% h?p????% ?`????% h	?P????% h
?@???? h
        ?0????%? f?1?I??^H??H???PTI??0
                                      @H???
@H??]	@?????fD??`UH-?`H??H??w]øH??t?]??`?????`UH-?`H??H??H??H???H?H??u]úH??t?]H?ƿ?`???? uUH???~???]?? ??@H?=? t?H??tU?`H????]?{????s???UH??H???P
                                       @??????`?????[
                                                     @?????l
                                                            @?n
                                                               @?[???H?E?H?}?u
??	H?U?H??P???H??H????????
                               @?f???H?E?H???
?????
u'H??P???H????????H?????????
                  uH??P???H???)?????t
????
    @????H??P???H??????H?H?ƿ?
                             @??N???H??P???H???O???H?H?ƿ?
                                                         @??+???H??P???H???????ƿ?
                                                                                 @??
                                                                                    ????
????H??P???H???b??????f.??AWA??AVI??AUI??ATL?%( UH?-( SL)?1?H??H??u???H??t?L??L??D??A??H??H9?u?H?[]A\A]A^A_Ðf.???H?H??WTFFFFFFFFHEREEEEEEEEEEEE1rholding_network_2002_sub.netHEREEEEEEEEEEEE2HEREEEEEEEEEEEE3The directed graph:Vertices: %li
Edges: %li
Directed: %i
0????|????Lq?????????D???
                         zRx
                           0???*zRx
                                  $ ????FJ
S                                         ??;*3$"D????XA?C
D???eB?E?E ?E(?H0?H8?M@l8A0A(B BB?0???0	@	@?
4                                                 h@
???o?@?@?@

` H@0	???o?@???o???o?@`?@?@?@?@?@?@@@@@GCC: (GNU) 4.8.5 20150623 (Red Hat 4.8.5-36)8@T@t@?@?@?@??@	@
H@
  h@
@@4 ?@
   @@
     @?
       @ 
         @```?``x`?`?
@?`????              `@.	@D(``z0	@?`??????
       @?``?`2@|`? x`G@
                       @Vd]	@XiH
                                    @_4
                                       @v?`???@?O
                                                 h@??`???x`?0`'|`3E?
@eU0
    @ex? crtstuff.c__JCR_LIST__deregister_tm_clones__do_global_dtors_auxcompleted.6355__do_global_dtors_aux_fini_array_entryframe_dummy__frame_dummy_init_array_entrypajek2.c__FRAME_END____JCR_END____GNU_EH_FRAME_HDR_DYNAMIC__init_array_end__init_array_start_GLOBAL_OFFSET_TABLE_printf@@GLIBC_2.2.5igraph_ecount_edata_IO_stdin_usedigraph_vcountmain__dso_handleigraph_cattribute_tableigraph_read_graph_pajekfclose@@GLIBC_2.2.5__TMC_END__fopen@@GLIBC_2.2.5igraph_i_set_attribute_table__data_startigraph_destroyputchar@@GLIBC_2.2.5__bss_startputs@@GLIBC_2.2.5__libc_csu_init__libc_csu_finiigraph_is_directed__libc_start_main@@GLIBC_2.2.5__gmon_start__.symtab.strtab.shstrtab.interp.note.ABI-tag.note.gnu.build-id.gnu.hash.dynsym.dynstr.gnu.version.gnu.version_r.rela.dyn.rela.plt.init.plt.got.text.fini.rodata.eh_frame_hdr.eh_frame.init_array.fini_array.jcr.dy@8#T@T 1t@t$D???o?@?Ns.comment
                     ?@?V?@?^???o?@?(k???o?@?z@?BH@H?h@h??@???@??4
                                                                  @4
                                                                    	?@
                                                                          @@
                                                                            ??
                                                                              @?
                                                                                4? 
                                                                                   @ 
                                                                                 ?`???x`x?`|? 0|-/      ??^

```

**Comment by ntamas** [comment]
Can you post the source code of the C program that you are trying to import the graph with? Preferably a minimal example that demonstrates the problem.

**Comment by cccfran** [comment]
I just changed the inputting line from the sample code `graphml.c`. Thanks.

```
#include <igraph.h>
#include <stdio.h>
#include <unistd.h>

void custom_warning_handler (const char *reason, const char *file,
			     int line, int igraph_errno) {
  printf("Warning: %s\n", reason);
}

int main(int argc, char **argv) {
  igraph_t g;
  igraph_error_handler_t* oldhandler;
  igraph_warning_handler_t* oldwarnhandler;
  int result;
  FILE *ifile, *ofile;

  /* GraphML */
  ifile=fopen("<filename>.graphml", "r");
  if (ifile==0) {
    return 10;
  }
  
  oldhandler=igraph_set_error_handler(igraph_error_handler_ignore);
  oldwarnhandler=igraph_set_warning_handler(custom_warning_handler);
  if ((result=igraph_read_graph_graphml(&g, ifile, 0))) {
    // maybe it is simply disabled at compile-time
    if (result == IGRAPH_UNIMPLEMENTED) return 77;
    return 1;
  }
  igraph_set_error_handler(oldhandler);

  fclose(ifile);
  
  printf("The directed graph:\n");
  printf("Vertices: %li\n", (long int) igraph_vcount(&g));
  printf("Edges: %li\n", (long int) igraph_ecount(&g));
  printf("Directed: %i\n", (int) igraph_is_directed(&g));
  igraph_write_graph_edgelist(&g, stdout);
  igraph_destroy(&g);
 
  
  return 0;
}
```

**Comment by ntamas** [comment]
The code that you have posted works perfectly fine on my machine:

```sh
tamas@tesseract:/tmp/
› gcc -o test test.c -I/Users/tamas/include/igraph -L/Users/tamas/lib -ligraph

tamas@tesseract:/tmp/
› ./test
The directed graph:
Vertices: 7
Edges: 5
Directed: 1
0 5
1 5
2 6
3 6
4 6
```

One vague idea that I have is that you have removed the setup of the attribute handler from the example, and since the input graph has attributes, this could cause problems down the line. I mean this one:

```c
igraph_i_set_attribute_table(&igraph_cattribute_table);
```

This should go in `main()` before you start loading graphs.

**Comment by cccfran** [comment]
I figured I added the line to read pajek not graphml. It works for .graphml now. Thanks a lot!

For pajek, it still fails to read in the output from R with error 
`Error at foreign.c:580 :Parse error in Pajek file, line 1 (syntax error, unexpected ALNUM), Parse error`
It only works after adding the vertices information as follows.
```
*Vertices 7
1 "1"
2 "2"
3 "3"
4 "4"
5 "5"
6 "6"
7 "7"
*Arcs
1 6 NaN
2 6 NaN
3 7 100
4 7 191
5 7 9
```
Just wondering it is a must to specify vertices information when reading .net?  Thanks again!
